File: leetcode/035-Search-Insert-Position/SearchInsertPosition_001.java
public class Solution {
    public int searchInsert(int[] A, int target) {
        int a = 0;
        int b = A.length - 1;
        int mid = 0;
        int loc = 0;
        
        if(A.length == 0){
            return 0;
        }
        
        if (A.length == 1){
            if(A[0] >= target){
                return 0;
            }
            else{
                return 1;
            }
        }
        
        while(b - a != 1){
            mid = (a+b)/2;
            loc = A[mid];
            if(loc == target){
                return mid;
            }
            else if(loc < target){
                a = mid;
                
            }
            else{
                b = mid;
            }
        }
        
        if(target <= A[a]){
            return 0;
        }
        else if(target > A[b]){
            return b+1;
        }
        else{
            return a+1;
        }
    }
}

File: lintcode/035-[DUP]-Search-Insert-Position/SearchInsertPosition_001.py
class Solution:
    """
    @param A : a list of integers
    @param target : an integer to be inserted
    @return : an integer
    """
    def searchInsert(self, A, target):
        # write your code here
        l, r = 0, len(A) - 1
        while l <= r:
            m = (l + r) / 2
            if A[m] == target:
                return m
            elif A[m] > target:
                r = m - 1
            else:
                l = m + 1
        return l

sl = Solution()
A = [1, 3, 5, 6]
target = [5, 2, 7, 0]
for t in target:
    print sl.searchInsert(A, t)
